### PR TITLE
check(controller): Check controller http server health probes.

### PR DESCRIPTION
**Pull request by johnsonshi**
check(controller): Check controller http server health probes.

Checks controller http server health probes for
the health of the osm-controller pods.

Signed-off-by: Johnson Shi <Johnson.Shi@microsoft.com>

### CODE REVIEW

**Comment by johnsonshi** [comment]
@SanyaKochhar // @draychev // @jaellio - Should we add some checks of `/metrics` from osm-controller's http server as well? I see lots of potentially useful metrics that we can obtain from `/metrics`. Example: https://pastebin.com/B06MeZwY

File: pkg/osm/osm_controller_http_server.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	url := fmt.Sprintf("%s%s", controllerHTTPServerURL, "/health/ready")
+	respBody, err := utils.GetResponseBody(url)
+	if err != nil {
+		return errors.Errorf("osm-controller health readiness check failed: %s", err)
+	}
+
+	if respBody != "Service is ready" {
+		return ErrorControllerNotReady
+	}
+	return nil
```
Comment (draychev) [inline pkg/osm/osm_controller_http_server.go:129]: Checking the response code (should be `200`) is sufficient.
We certainly should not be checking the response body. Response body should mostly be for humans to understand things went ok. Machines should be checking the response codes and headers in this particular scenario.

**Review by draychev — CHANGES_REQUESTED** [review]
@johnsonshi I recommend we only check the response headers and status code - this should be sufficient.
I don't think we should be checking the response body and matching against specific text. All we care about is response code `200` and potentially some interesting headers.

File: pkg/osm/controller/types.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package controller
+
+import "github.com/openservicemesh/osm-health/pkg/logger"
+
+var log = logger.New("control-plane/osm-controller/http-server")
```
Comment (johnsonshi) [inline pkg/osm/controller/types.go:5]: @draychev is this a suitable log name?

**Review by draychev — APPROVED** [review]

**Pull request merged**